### PR TITLE
metrics: add proxy error counter

### PR DESCRIPTION
**Pull request by Choraden**

### CODE REVIEW

**Comment by Choraden** [comment]
This approach bases on the assumption that proxy will set `ErrorReason` in request's context. It does not report `denied` or `unauthorized` requests as `Proxy.abortIf` method lacks prometheus modifiers.

**Comment by Choraden** [comment]
New approach makes errorResponse function a proxy's method and reports metrics there.

**Comment by mmatczuk** [comment]
I'm not sure if we have to add `_error` suffix to label.
I'd suggest renaming metrics to httpProxyMetrics.
I'd suggest renaming http_errors.go to http_proxy_errors and similar with metrics.

File: http_error.go [hunk 1]
```diff
@@ -72,36 +75,40 @@ func handleNetError(_ *http.Request, err error) (code int, msg string) {
 			code = http.StatusBadGateway
 			msg = "Failed to connect to remote host"
 		}
+		label = "net_" + netErr.Op + "_error"
```
[the remark this answers is not in the document]
Comment (Choraden) [reply http_error.go:78]: `Err` can contain several variables e.g. `ip`, `hostname`. I doubt if filtering all these things is worth it. One can always check logs if increased error activity is reported.

**Comment by mmatczuk** [comment]
LGTM

**Pull request merged**